packages/@aws-cdk/aws-apigateway/lib/restapi.ts

```diff
@@ -403,9 +403,11 @@ export abstract class RestApiBase extends Resource implements IRestApi {
     });
 
     resource.node.addDependency(apiResource);
+
+    return resource;
   }
 
-  protected configureDeployment(props: RestApiOptions) {
```

Comment:
The fix currently only configures a 'depends on' for the auto-created stage. This will need to be extended to apply to all stages, including the one that users will create.

I would suggest changing to add this dependency within the constructor of `Stage` and retrieving the `CfnAccount` using an internal getter. 
Take a look at some of the method in `restapi.ts` that are marked `@internal` to see how this can be done.

Reply:
Thanks will check into that.